lib/gnat/consumer_supervisor.ex

```diff
@@ -148,4 +188,33 @@ defmodule Gnat.ConsumerSupervisor do
         wait_for_empty_task_supervisor(state)
     end
   end
+
+  defp initialize_as_manual_consumer(state, connection_pid) do
+    subscriptions = Enum.map(state.subscription_topics, fn(topic_and_queue_group) ->
+      topic = Map.fetch!(topic_and_queue_group, :topic)
+      {:ok, subscription} = case Map.get(topic_and_queue_group, :queue_group) do
+        nil -> Gnat.sub(connection_pid, self(), topic)
+        queue_group -> Gnat.sub(connection_pid, self(), topic, queue_group: queue_group)
+      end
+      subscription
+    end)
+
+    %{ state | subscriptions: subscriptions }
+  end
+
+  defp initialize_as_microservice(state, connection_pid) do
+    {:ok, responder_pid} = ServiceResponder.start_link(%{ state | connection_pid: connection_pid })
+    endpoints = get_in(state, [:service_config, :endpoints])
+
+    subscriptions = Enum.map(endpoints, fn(ep) ->
+      subject = ServiceResponder.derive_subscription_subject(ep)
+      queue_group = Map.get(ep, :queue_group, @default_service_queue_group)
+      {:ok, subscription} = Gnat.sub(connection_pid, self(), subject, queue_group: queue_group)
+
+      subscription
+    end)
+
+    %{state | svc_responder_pid: responder_pid, subscriptions: subscriptions }
```

Comment:
I'm part way through reviewing this and I have to jump on a work call soon. I'm trying to wrap my head around the use-case for the `ets` table and `responder_pid`. But I wanted to provide some of the early feedback before my work day gets busy

Reply:
Looking forward to your feedback. The `responder_pid` is to deal with the fact that whenever a consumer supervisor is managing a service, it creates a new child (`ServiceResponder`) that is responsible for handling the protocol requests like `PING`, `STATS`, and `INFO`. This responder maintains this automatically and uses the ets table as a way to allow both the `Gnat.Services.Server` and the responder to access the stat counters for processing time, errors, and success